app/uk/gov/hmrc/gform/views/form/snippets/field_template_text_total.scala.html

```diff
@@ -29,6 +29,18 @@
 @import uk.gov.hmrc.gform.validation.{FormFieldValidationResult, ValidationUtil}
 @(fieldValue: FormComponent, text: Text, prepop: String, validationResult: Option[FormFieldValidationResult], index: Int, sectionTitle: String)
 
+@currentVal = @{ val current = validationResult.flatMap(_.getCurrentValue).getOrElse(prepop)
+    val zero = BigDecimal("0")
+    val currentAmount = BigDecimal(current)
+    if (currentAmount < zero){
+        val finalNegativeAmount = "-£" + currentAmount.abs.toString()
+        finalNegativeAmount
+    } else {
+        val finalPositiveAmount = "£" + current
+        finalPositiveAmount
+    }
+}
```

Comment:
This seems to be broken in two ways.

1. We are adding `£` even for non Sterling values. (I do not see equivalent logic of removed line 53)
2. In other places of application we took approach to use `NumberFormat.getCurrencyInstance(Locale.UK)` to format Sterling values. We should use it here as well. See [TextFormatter.scala](https://github.com/hmrc/gform-frontend/blob/master/app/uk/gov/hmrc/gform/views/summary/TextFormatter.scala#L33) for inspiration.

Reply:
@josef-vlach  I believe this should fix all reported issues?